### PR TITLE
Bump LLVM to 1ebf1afb4ff

### PR DESCRIPTION
**Pull request by nandor**
- `AttrDef` and `TypeDef` custom assembly has changed: https://reviews.llvm.org/D121505
- `FunctionOpInterface::getType()` renamed to `getFunctionType()` https://reviews.llvm.org/D121762

### CODE REVIEW

File: include/circt/Dialect/FIRRTL/Passes.td [hunk 1]
```diff
@@ -98,6 +98,12 @@ def Inliner : Pass<"firrtl-inliner", "firrtl::CircuitOp"> {
     ```
   }];
   let constructor = "circt::firrtl::createInlinerPass()";
+  let statistics = [
+    Statistic<"numRemoved", "num-removed-modules",
+      "Number of dead modules erased">,
+    Statistic<"numInlined", "num-inlined-instances",
+      "Number of instances inlined">
+  ];
```
Comment (youngar) [inline include/circt/Dialect/FIRRTL/Passes.td:106]: Did this accidentally get sucked in?

Comment (nandor) [reply include/circt/Dialect/FIRRTL/Passes.td]: Yes, deleted.

File: lib/Dialect/ESI/ESITypes.cpp [hunk 2]
```diff
@@ -21,6 +21,9 @@
 using namespace circt;
 using namespace circt::esi;
 
+/*
```
Comment (teqdruid) [inline lib/Dialect/ESI/ESITypes.cpp:24]: You can delete this code. I assume it didn't compile with the new LLVM and switching to `assemblyFormat` was the simpler thing to do?

**Review by teqdruid — APPROVED** [review]
I'm approving the ESI and MSFT changes -- they look good. Thanks!

File: lib/Dialect/Calyx/CalyxOps.cpp [hunk 3]
```diff
@@ -786,7 +781,6 @@ static LogicalResult isCombinational(Value value, GroupInterface group) {
 /// Verifies a combinational group may contain only combinational primitives or
 /// perform combinational logic.
 LogicalResult CombGroupOp::verify() {
-
```
Comment (seldridge) [inline lib/Dialect/Calyx/CalyxOps.cpp:789]: Supreme nit: whitespace change.

File: lib/Dialect/Comb/CombOps.cpp [hunk 4]
```diff
@@ -181,22 +181,21 @@ LogicalResult ReplicateOp::verify() {
 // Variadic operations
 //===----------------------------------------------------------------------===//
 
-static LogicalResult verifyUTVariadicOp(Operation *op) {
+static LogicalResult verifyUTBinOp(Operation *op) {
   if (op->getOperands().empty())
     return op->emitOpError("requires 1 or more args");
-
   return success();
 }
```
Comment (seldridge) [inline lib/Dialect/Comb/CombOps.cpp:188]: Mega nit: this appears to be just a name-change as opposed to anything functional? You can consider landing this separately. This is not a blocker.

File: lib/Dialect/ESI/ESITypes.cpp [hunk 5]
```diff
@@ -21,6 +21,9 @@
...
+TODO BUMP
+
 Type ChannelPort::parse(AsmParser &p) {
```
Comment (seldridge) [inline lib/Dialect/ESI/ESITypes.cpp]: This is unfortunate, but it sounds like @teqdruid is fine with deleting it. 😌

File: lib/Dialect/FIRRTL/FIRRTLOps.cpp [hunk 6]
```diff
@@ -278,7 +278,7 @@ LogicalResult CircuitOp::verify() {
 
     // Check that this extmodule's defname does not conflict with
     // the symbol name of any module.
-    auto *collidingModule = lookupSymbol(defname.getValue());
+    auto collidingModule = lookupSymbol(defname.getValue());
```
Comment (seldridge) [inline lib/Dialect/FIRRTL/FIRRTLOps.cpp:281]: Mega-nit: this can land separately if you want.

File: lib/Dialect/ESI/ESITypes.cpp (hunk 2 above)
Comment (nandor) [reply lib/Dialect/ESI/ESITypes.cpp]: I didn't even find the point where `ChannelPort` is defined - I assume it's now using some default `assemblyFormat`.

File: lib/Dialect/HW/HWAttributes.cpp [hunk 7]
```diff
@@ -161,21 +161,6 @@ void OutputFileAttr::print(AsmPrinter &p) const {
...
-      p.parseGreater())
-    return Attribute();
-  auto *context = p.getContext();
-  return FileListAttr::get(context, filename);
-}
-
-void FileListAttr::print(AsmPrinter &p) const {
-  p << "<" << getFilename() << ">";
-}
-
```
Comment (seldridge) [inline lib/Dialect/HW/HWAttributes.cpp:178]: This cleanup is awesome. 💯

File: lib/Dialect/SV/SVOps.cpp [hunk 8]
```diff
@@ -212,7 +211,7 @@ void LocalParamOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
 LogicalResult LocalParamOp::verify() {
   // Verify that this is a valid parameter value.
   return hw::checkParameterInContext(
-      value(), (*this)->getParentOfType<hw::HWModuleOp>(), (*this));
+      value(), (*this)->getParentOfType<hw::HWModuleOp>(), *this);
```
Comment (seldridge) [inline lib/Dialect/SV/SVOps.cpp:214]: Supreme-nit: always feel free to land these types of fixes separately. (Not a blocker.)

File: lib/Dialect/SV/SVOps.cpp [hunk 9]
```diff
@@ -1131,7 +1130,29 @@ LogicalResult IndexedPartSelectInOutOp::inferReturnTypes(
...
+          getType().cast<InOutType>().getElementType().dyn_cast<IntegerType>())
+    resultWidth = resType.getWidth();
+  else
+    return emitError("result element type must be Integer");
+
+  if (opWidth > inputWidth)
+    return emitError("slice width should not be greater than input width");
+  if (opWidth != resultWidth)
+    return emitError("result width must be equal to slice width");
+  return success();
```
Comment (seldridge) [inline lib/Dialect/SV/SVOps.cpp:1155]: I eyeballed this and it looks like the exact same logic, just a nice refactor. 👍

**Review by seldridge — APPROVED** [review]
Awesome!  This is a lot of tedious updates.  Thank you for tackling it.

File: lib/Dialect/SV/SVOps.cpp (hunk 9 above)
Comment (nandor) [reply lib/Dialect/SV/SVOps.cpp:1155]: I found that not enough code was shared to warrant the dynamic type checks.

**Pull request merged**